core/src/main/java/com/datastrato/gravitino/listener/SupportsAsync.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener;
+
+import com.datastrato.gravitino.annotation.Evolving;
+
+/**
+ * This interface is designed for event listeners that process events asynchronously. Implementing
+ * this interface allows an event listener to specify whether it processes events using a shared
+ * dispatcher or its own separate dispatcher.
+ *
+ * <p>Use of a shared dispatcher can help in efficiently managing system resources by sharing a
+ * common event-processing queue across multiple listeners. Conversely, opting for a separate
+ * dispatcher allows for more isolated and potentially more customizable event processing, at the
+ * cost of additional resource overhead.
+ *
+ * <p>Implementers of this interface should consider the trade-offs between resource efficiency and
+ * processing isolation when deciding on the type of dispatcher to use.
+ */
+@Evolving
+public interface SupportsAsync {
+  /**
+   * Indicates whether the event listener processes events using a shared dispatcher or its own
+   * separate dispatcher.
+   *
+   * <p>A shared dispatcher implies that events will be processed in a common queue shared among
+   * multiple listeners, potentially leading to more efficient resource usage. A separate dispatcher
+   * implies that the listener has its own dedicated event-processing queue, allowing for more
+   * isolated processing but at the expense of additional resource consumption.
+   *
+   * @return {@code true} if the event listener uses a shared dispatcher, {@code false} if it uses a
+   *     separate dispatcher.
+   */
+  default boolean isSharedDispatcher() {
+    return true;
```

Comment:
I would suggest to define a `enum Mode {ISOLATED, SHARED}` with default method as `default Mode asyncMode() { return SHARED}`. Carefully explain the two modes with Javadoc.

The current design has two slight problems:
1. It introduces the concept of "dispatcher" in API, but what's relation to `SupportsAsync`, it makes the users hard to understand and explain.
2. It basically lets user to choose either shared or isolated, using true or false is a little weird.

Reply:
done